packages/sanity/src/_internal/cli/actions/schema/schemaListAction.ts

```diff
@@ -0,0 +1,141 @@
+import {type CliCommandArguments, type CliCommandContext, type CliOutputter} from '@sanity/cli'
+import {type SanityDocument} from '@sanity/client'
+import chalk from 'chalk'
+import {size, sortBy, uniqBy} from 'lodash'
+
+import {type ManifestWorkspaceFile} from '../../../manifest/manifestTypes'
+import {
+  getManifestPath,
+  readManifest,
+  SCHEMA_STORE_ENABLED,
+  throwIfProjectIdMismatch,
+} from './storeSchemasAction'
+
+export interface SchemaListFlags {
+  'json': boolean
+  'id': string
+  'manifest-dir': string
+}
+
+type PrintSchemaListArgs = {
+  schemas: SanityDocument[]
+  output: CliOutputter
+  dataset: string
+  projectId: string
+  path: string
+}
+
+export const SANITY_WORKSPACE_SCHEMA_TYPE = 'sanity.workspace.schema'
+
+const printSchemaList = ({
+  schemas,
+  output,
+}: Omit<PrintSchemaListArgs, 'path' | 'dataset' | 'projectId'>) => {
+  const ordered = sortBy(
+    schemas.map(({_createdAt: createdAt, _id: id, workspace}) => {
+      return [id, workspace.name, workspace.dataset, workspace.projectId, createdAt].map(String)
+    }),
+    ['createdAt'],
+  )
+  const headings = ['Id', 'Workspace', 'Dataset', 'ProjectId', 'CreatedAt']
+  const rows = ordered.reverse()
+
+  const maxWidths = rows.reduce(
+    (max, row) => row.map((current, index) => Math.max(size(current), max[index])),
+    headings.map((str) => size(str)),
+  )
+
+  const printRow = (row: string[]) => row.map((col, i) => `${col}`.padEnd(maxWidths[i])).join('   ')
+
+  output.print(chalk.cyan(printRow(headings)))
+  rows.forEach((row) => output.print(printRow(row)))
+}
+
+export default async function schemaListAction(
+  args: CliCommandArguments<SchemaListFlags>,
+  context: CliCommandContext,
+): Promise<void> {
+  if (!SCHEMA_STORE_ENABLED) {
+    return
+  }
+
+  const flags = args.extOptions
+  if (typeof flags.id === 'boolean') throw new Error('Schema ID is empty')
+  if (typeof flags['manifest-dir'] === 'boolean') throw new Error('Manifest directory is empty')
+
+  const {apiClient, output} = context
+  const client = apiClient({
+    requireUser: true,
+    requireProject: true,
+  }).withConfig({apiVersion: 'v2024-08-01'})
+
+  const projectId = client.config().projectId
+  const dataset = client.config().dataset
+
+  if (!projectId || !dataset) {
+    output.error('Project ID and dataset must be defined.')
+    return
+  }
+
+  const manifestDir = flags['manifest-dir']
+  const manifestPath = getManifestPath(context, manifestDir)
+  const manifest = await readManifest(manifestPath, context)
+
+  // Gather all schemas
```

Comment:
Nit: Noisy comment (delete imo)

Edit: I see there are a lot of this, and Its just your style. I'd rather we didnt litter the code with comments that repeats what the code is doing, especially when its clear on the very next line (maybe not here).

If this variable was named schemaResults you dont need the comment. Ect.

I'll leave it up to you, but I cant remember seeing this style often in the sanity code-base elsewhere; imo prefer comments for "whys or really complicated, im sorry" stuff 😅 

If you want to add meatier TS docs comments at the top of fuctions though, im all for it.